geonode/harvesting/harvesters/wms.py

```diff
@@ -104,66 +136,179 @@ def check_availability(self, timeout_seconds: typing.Optional[int] = 5) -> bool:
             result = True
         return result
 
+    def get_geonode_resource_type(self, remote_resource_type: str) -> ResourceBase:
+        """Return resource type class from resource type string."""
+        # WMS just have Layer type on it's resource.
+        # So whatever remote_resource_type it is, it always return Layer.
+        return Layer
+
     def get_resource(
             self,
-            resource_unique_identifier: str,
-            remote_resource_type: str,
-            harvesting_session_id: typing.Optional[int] = None
-    ) -> typing.Optional[resourcedescriptor.RecordDescription]:
-        params = self._base_wms_parameters
+            harvestable_resource: "HarvestableResource",  # noqa
+            harvesting_session_id: int
+    ) -> typing.Optional[base.HarvestedResourceInfo]:
+        resource_unique_identifier = harvestable_resource.unique_identifier
+        data = self._get_data()
+        for layer in data['layers']:
```

Comment:
IMO it would make the code more readable if instead of this nested `for` loop with an `if` inside of it (adds a lot of visual noise due to the indentation) you would get the relevant layer first and then process it. Something like:

```python
try:
    relevant_layer = [layer for layer in data["layers"] if layer["name"] == resource_unique_identifier][0]
except IndexError:
    result = None
    logger.exception(f"Could not find resource {resource_unique_identifier!r}")
else:
    # now do stuff with the `relevant_layer`
```

Also, in my opinion (but I guess this is a hot topic) it would be preferable to use a singe `return` statement, instead of multiple returns - I think it makes the code easier to read.